scipy/sparse/_matrix_io.py

```diff
@@ -1,15 +1,15 @@
 import numpy as np
-import scipy.sparse
+import scipy as sp
 
-__all__ = ['save_npz', 'load_npz']
+__all__ = ['save_npz', 'load_npz', 'load_npz_array']
 
 
 # Make loading safe vs. malicious input
 PICKLE_KWARGS = dict(allow_pickle=False)
 
 
 def save_npz(file, matrix, compressed=True):
```

Comment:
Could we include metadata to indicate which container was stored? It may still make sense to have a flag / example to show how to force a specific format.

```
In [51]: np.savez('/tmp/foo.npz', __format__='array')

In [52]: np.load('/tmp/foo.npz')['__format__']
Out[52]: array('array', dtype='<U5')
```

Reply:
I added a key-value pair to the npz file with key `__is_array` and value `array` or `matrix`. This gets concatenated with `self.format` to create the class constructor `sp.sparse.<format>_<array or matrix>`.
Other choices are certainly possible. I also added an example which takes the output based on `__is_array` and changes it by feeding it into a `csr_array` constructor.